Src/FluentAssertions/CallerIdentification/IHandler.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace FluentAssertions.CallerIdentification
+{
+    public interface IHandler
```

Comment:
🔧 Thinking about the algorithm, it seems to me that this is really represents a parsing strategy, so I suggest to rename this to `IParser` or `IParsingStrategy`
🔧 I would also add some docs and explain its purpose, especially about the fact that these handlers are state-full and created once per parsing task.

Reply:
Done